README.md

```diff
@@ -14,71 +16,252 @@ Takeaway Challenge
+- Use the Gemfile to manage your gems
+- Make sure that your Takeaway is thoroughly tested and that you use mocks and/or stubs, as necessary to not to send texts when your tests are run
+- However, if your Takeaway is loaded into IRB and the order is placed, the text should actually be sent
+- Note that you can only send texts in the same country as you have your account. I.e. if you have a UK account you can only send to UK numbers.
+### Advanced  (have a go if you're feeling adventurous):
+* Implement the ability to place orders via text message.
+
+* A free account on Twilio will only allow you to send texts to "verified" numbers. Use your mobile phone number, don't worry about the customer's mobile phone.
+
+> :warning: **WARNING:** think twice before pushing your **mobile number** or **Twilio API Key** to a public space like GitHub :eyes:
+>
+> :key: Think about security and how you can keep your private information secret. Explore environment variables.
+
+## My approach
+
+### 1. Modelling
+
+  - I start by writing a domain model for a user story.
+  - Then I draw a class diagram.
+  - Do just enough design upfront to start.
 
-* Challenge time: rest of the day and weekend, until Monday 9am
-* Feel free to use google, your notes, books, etc. but work on your own
-* If you refer to the solution of another coach or student, please put a link to that in your README
-* If you have a partial solution, **still check in a partial solution**
-* You must submit a pull request to this repo with your code by 9am Monday morning
+### 2. TDD
 
-Task
------
+  - Start with a test for the smallest meaningful feature.
+  - Writea test and watch it fail before writing code.
+  - Only write code to fix the test fail message I can see.
+  - Include edge cases in tests.
+  - Look for an opportunity to refactor the code after getting to green.
+  - Refactore verbose tests after the code is working.
 
-* Fork this repo
-* Run the command 'bundle' in the project directory to ensure you have all the gems
-* Write a Takeaway program with the following user stories:
+### Problem solving 
+  - Try to understand the problem by diagramming, or writing pseudo-code.
+  - Break the problem down into smaller problems.
+  - Verbalise my thought process when solving a problem.
+
+## User Story 1
 
 ```
 As a customer
 So that I can check if I want to order something
 I would like to see a list of dishes with prices
+```
+From the users perspective in IRB:
+
+```
+order = Order.new
+=> #<Order:0x00007f863783fda0 @menu={:fish=>5, :chips=>3, :salad=>2}> 
+order.menu
+=> {:fish=>"£5", :chips=>"£3", :salad=>"£2"} 
+```
 
+## User Story 2
+
+```
 As a customer
 So that I can order the meal I want
 I would like to be able to select some number of several available dishes
+```
+From the users perspective in IRB:
+```
+order = Order.new
+=> #<Order:0x00007f863783fda0 @menu={:fish=>5, :chips=>3, :salad=>2}> 
+order.update("1 fish, 3 salad, 2 chips")
```

Comment:
The method name `update` is a bit confusing. Without reading the code, I'm not intuitively sure what would happen if I did
`order.update("1 fish, 3 salad, 2 chips")`
followed by
`order.update("1 fish, 1 salad, 1 chips")`.
Would I end up with 2 fish, 4 salad, and 3 chips or 1 fish, 1 salad, 1 chips. I wonder if calling the method name `add` would make this clearer here?